bookdown/08-use-cases.Rmd

```diff
@@ -1,3 +1,405 @@
+```{r}
+tsk_nozip = tsk$clone()$select(tsk$feature_names[1:19])
+# Get the learner
+lrn = mlr_learners$get("regr.rpart")
+# And train on the task
+lrn$train(tsk_nozip, row_ids = train.idx)
+```
+
+```{r, width = 10, height = 10}
+plot(lrn$model)
+text(lrn$model)
+```
+
+The learned tree relies on several variables in order to distinguish between cheaper and pricier houses.
+The features we split along seem to be **grade**, **sqft_living**, but also some features related to the area (longitude and latitude).
+
+We can visualize the price across different regions in order to get more info:
+
+```{r, message = FALSE}
+# Load the ggmap package in order to visualize on a map
+library(ggmap)
+
+# And create a quick plot for the price
+qmplot(long, lat, maptype = "watercolor", color = log(price),
+  data = house_sales_prediction[train.idx[1:3000],]) +
+  scale_colour_viridis_c()
+
+# And the zipcode
+qmplot(long, lat, maptype = "watercolor", color = zipcode,
+  data = house_sales_prediction[train.idx[1:3000],]) + guides(color = FALSE)
+```
+
+We can see that the price is clearly associated with the zipcode when comparing the
+two plots. 
+As a result, we might want to indeed use the **zipcode** column in our future endeavours.
+
+
+### A first baseline: Decision Tree
+
+After getting an initial idea for our data, we might want to construct a first 
+baseline, in order to see what a simple model already can achieve.
+
+We use `resample` with `3-fold Cross-Validation` on our training data in order to get a reliable estimate of the algorithm's performance on future data.
+
+
+Before we start with defining and training learners, we create a `r ref("Resampling")`
+in order to make sure that we always compare on exactly the same data.
+
+```{r}
+library(mlr3learners)
+cv3 = mlr_resamplings$get("cv3")
+cv3$instantiate(tsk$clone()$filter(train.idx))
+```
+
+For the cross-validation we only use the **training data** by cloning the task and selecting only observations from the training set.
+
+```{r}
+lrn_rpart = mlr_learners$get("regr.rpart")
+res = resample(task = tsk$clone()$filter(train.idx), lrn_rpart, cv3)
+res$performance("regr.mse")
+sprintf("RMSE of the simple rpart: %s", round(sqrt(res$aggregate()), 2))
+```
+
+### Many Trees: Random Forest
+
+We might be able to improve upon the **RMSE**.
+We first load the `mlr3learners` package, which contains **ranger**.
+
+```{r}
+lrn_ranger = mlr_learners$get("regr.ranger", param_vals = list(num.trees = 15L))
+res = resample(task = tsk$clone()$filter(train.idx), lrn_ranger, cv3)
+res$performance("regr.mse")
+sprintf("RMSE of the simple ranger: %s", round(sqrt(res$aggregate()), 2))
+```
+
+Oftentimes tuning **RandomForest** methods does not increase predictive performances drastically. If time permits, it can nonetheless lead to improvements and should thus be tried out. In this case, we resort to tune a different kind of model: **Gradient Boosted Decision Trees** from the package `r cran_pkg("xgboost")` instead.
```

Comment:
```suggestion
Often tuning **RandomForest** methods does not increase predictive performances substantially.
If time permits, it can nonetheless lead to improvements and should thus be performed. 
In this case, we resort to tune a different kind of model: **Gradient Boosted Decision Trees** from the package `r cran_pkg("xgboost")`.
```